### PR TITLE
Feature/more forgiving batch wait

### PR DESCRIPTION
**Pull request by prescod**
# Critical Changes

# Changes

Previously, CCI would check that a batch job was completed correctly based on whether it both reported no error message and also had a TotalJobItems=JobItemsProcessed. On extremely large jobs with lots of Ajax, this seems like too conservative of a strategy. Sometimes the Salesforce Platform returns JobItemsProcessed<TotalJobItems but otherwise it seems that job succeeds and there is no serious reason to be concerned. In this case, CCI will now report a warning rather than an error and other processes can continue.

# Issues Closed

### CODE REVIEW

File: cumulusci/tasks/apex/batch.py [hunk 1]
```diff
@@ -33,9 +32,28 @@ def _run_task(self):
...
+            for key, value in self.batch.items()
+            if key
+            in [
+                "Id",
+                "Status",
+                "ExtendedStatus",
+                "NumberOfErrors",
+                "JobItemsProcessed",
+                "TotalJobItems",
+            ]
```
Comment (davisagli) [inline cumulusci/tasks/apex/batch.py]: Not really consequential, but probably makes more sense to use a set literal here?

File: cumulusci/tasks/apex/batch.py [hunk 2]
```diff
@@ -33,9 +32,28 @@ def _run_task(self):
...
+        }
         if not self.success:
-            self.logger.info("There were some batch failures.")
-            raise SalesforceException(self.batch["ExtendedStatus"])
+            self.logger.info("There have been some batch failures.")
+            self.logger.info("Error values:")
+            self.logger.info(repr(vals))
+            raise SalesforceException("There were import errors", vals)
```
Comment (davisagli) [inline cumulusci/tasks/apex/batch.py]: This is going to print something like:

    cumulusci.core.exceptions.SalesforceException: ('There were import errors', "{'Id': 'asdf', etc... }")

Did you mean to format this somehow?

File: cumulusci/tasks/apex/batch.py [hunk 3]
```diff
@@ -33,9 +32,28 @@ def _run_task(self):
...
+        elif not self.done_for_sure:
+            self.logger.info("The final record counts do not add up.")
+            self.logger.info("This is probably related to W-1132237")
```
Comment (davisagli) [inline cumulusci/tasks/apex/batch.py]: We should probably link to the known issue (which is public) rather than an internal work item.

Comment (prescod) [reply cumulusci/tasks/apex/batch.py]: Google turns up the public page when you search for W-1132237

Comment (davisagli) [reply cumulusci/tasks/apex/batch.py]: I forgot those are referenced publicly. But why not save someone the googling and link to https://success.salesforce.com/issues_view?id=a1p300000008XvaAAE

File: cumulusci/tasks/apex/batch.py (hunk 1 above)
Comment (prescod) [reply cumulusci/tasks/apex/batch.py]: f0fa6cc2775f05b282dd44d730538e39219b11f6

File: cumulusci/tasks/apex/batch.py (hunk 2 above)
Comment (prescod) [reply cumulusci/tasks/apex/batch.py]: I didn't feel strongly about this, but PEP 352 advocates for a single string argument, so I've done that in e2368baa9e03caeb68b394c971de85d4836d78a7

File: cumulusci/tasks/apex/batch.py (hunk 3 above)
Comment (prescod) [reply cumulusci/tasks/apex/batch.py]: 99f6708a

**Comment by davisagli** [comment]
@prescod can you update this PR description to include a description of the change that's suitable to put in the release notes?

**Pull request merged**